### ISSUE TITLE
ft_checkdata timelock runs into trouble when a trial's beg/end sample has two equal nearest

### ISSUE THREAD

**Issue by matsvanes** [fieldtrip/fieldtrip#1072]
Consider the attached data (selected the trials that defined the time axis (earliest begin time point and latest end time point) and the erroneous trial). 
If I run `ft_checkdata(data, 'datatype', 'timelock')` I get the following error:
```
Subscripted assignment dimension mismatch.

Error in ft_checkdata>raw2timelock (line 1495)
    tmptrial(i,:,begsmp(i):endsmp(i)) = data.trial{i};

Error in ft_checkdata (line 449)
      data = raw2timelock(data);
```

The piece of code  where this error occurs (lines 1523-1529):
```
begsmp = nan(ntrial, 1);
endsmp = nan(ntrial, 1);
  for i=1:ntrial
    begsmp(i) = nearest(time, data.time{i}(1));
    endsmp(i) = nearest(time, data.time{i}(end));
    tmptrial(i,:,begsmp(i):endsmp(i)) = data.trial{i};
  end
```

because `data.time{i}(1)` has _two_ nearest at `time`:
```
begtim =  data.time{i}(1);
abs(begtim-time(begsmp(i)))-abs(begtim-time(begsmp(i)+1))
ans =

     0
```

while 

```
endtim =  data.time{i}(end);
abs(endtim-time(endsmp(i)))-abs(endtim-time(endsmp(i)-1))
ans =ans =

    -8.881784197001252e-16
```
So trial _i_ has a time axis that is exactly in between the `time` as defined by the 'longest' trials. Due to numerical imprecision this doesn't present itself at every sample, leading to inconsistencies.
This issue presented itself during ft_timelockanalysis. The issue was absent when I analysed the data before, because  `ft_checkdata(data, 'datatype', 'timelock')` was added fairly recently commit #(b8a449568438dca5fbe6c02d0bcb99f952aa94a1 ).
[issue_checkdata_timelock.zip](https://github.com/fieldtrip/fieldtrip/files/3038399/issue_checkdata_timelock.zip)

**Comment by matsvanes** [comment]
@robertoostenveld This is the issue with rounding errors I talked about.

**Comment by matsvanes** [comment]
Fixed in #1084
